### DISCUSSION TITLE
Number of comparisons with highly redundant data

### DISCUSSION THREAD

**Discussion by fbilhaut** [moj-analytical-services/splink#1634] [Q&A]
Hi everyone,

Just a very simple question about what happens when using Splink on highly redundant data.

Let's say that the data set contains (beside other ones) `N` entries with exactly the same data in two columns `A` and `B`, and that I use the following blocking rule: `l.A = r.A AND l.B = r.B`.

As far as I understand : all pairwise comparisons will be generated which leads to a list of `C(n,2) = N(N-1)/2` comparisons to be done.

So for example if the dataset contains `10K` entries with the same values for `A` and `B`, it would generate something close to `50M` comparisons just for these entries. Which becomes quickly unmanageable, especially if you happen to have a bunch of groups of this kind.

Of course, one can easily come up with various ways to protect Splink from being fed with this kind of "pathological entries", but I was wondering if it provides something by itself beside writing some more restrictive blocking rules.

Thanks a lot

**Comment by RobinL** [comment]
Your understanding is correct.

Are you saying they're 'pathological' because they're obviously matches?  (i.e. you're feeding the model data that is obviously duplicates?)

Splink doesn't provide a way of avoiding this at the moment.  I guess because they're not necessarily pathological.  A could be first_name and B could be surname, and you might just have 10K John Smiths (mostly different people)  in a large dataset

There have been a couple of discussions before about:
1. Deduping the obvious dupes prior to deduping.  So in that case it'd be like `df.drop_duplicates(subset=['first_name', 'surname']`.  But a problem with this is how to then re-link in the obvious dupes
2. Rules for 'definite no' and possibly 'definite yes' criteria.  See here https://github.com/moj-analytical-services/splink/issues/806

Do you have any thoughts about what the logic would be to achieve what you're after?
